Přehled dat Olist - cíl: zjistit, jak vypadají tabulky, jak jsou velké a jaká obsahují data.

In [8]:
#Načtení knihoven a cest k datům
import pandas as pd
from pathlib import Path

DATA = Path("../data/raw")

In [9]:
#Přehled všech souborů v adresáři
for file in sorted(DATA.glob("*.csv")):
    df = pd.read_csv(file)
    raws, cols = df.shape
    print(f"{file.name}: {raws} raws, {cols} columns")
    

olist_customers_dataset.csv: 99441 raws, 5 columns
olist_geolocation_dataset.csv: 1000163 raws, 5 columns
olist_order_items_dataset.csv: 112650 raws, 7 columns
olist_order_payments_dataset.csv: 103886 raws, 5 columns
olist_order_reviews_dataset.csv: 99224 raws, 7 columns
olist_orders_dataset.csv: 99441 raws, 8 columns
olist_products_dataset.csv: 32951 raws, 9 columns
olist_sellers_dataset.csv: 3095 raws, 4 columns
product_category_name_translation.csv: 71 raws, 2 columns


In [20]:
geolocation = pd.read_csv(DATA / "olist_geolocation_dataset.csv")
geolocation.head()

,geolocation_zip_code_prefix,geolocation_lat,geolocation_lng,geolocation_city,geolocation_state
0,1037,-23.545621,-46.639292,sao paulo,SP
1,1046,-23.546081,-46.644820,sao paulo,SP
2,1046,-23.546129,-46.642951,sao paulo,SP
3,1041,-23.544392,-46.639499,sao paulo,SP
4,1035,-23.541578,-46.641607,sao paulo,SP


In [10]:
#Načtení objednávek a náhled prvních 5 řádků
orders = pd.read_csv(DATA / "C:\\Users\\svobo\\Projekty\\olist-analyza\\data\\raw\\olist_orders_dataset.csv")
orders.head()

,order_id,customer_id,order_status,order_purchase_timestamp,order_approved_at,order_delivered_carrier_date,order_delivered_customer_date,order_estimated_delivery_date
0,e481f51cbdc54678b7cc49136f2d6af7,9ef432eb6251297304e76186b10a928d,delivered,2017-10-02 10:56:33,2017-10-02 11:07:15,2017-10-04 19:55:00,2017-10-10 21:25:13,2017-10-18 00:00:00
1,53cdb2fc8bc7dce0b6741e2150273451,b0830fb4747a6c6d20dea0b8c802d7ef,delivered,2018-07-24 20:41:37,2018-07-26 03:24:27,2018-07-26 14:31:00,2018-08-07 15:27:45,2018-08-13 00:00:00
2,47770eb9100c2d0c44946d9cf07ec65d,41ce2a54c0b03bf3443c3d931a367089,delivered,2018-08-08 08:38:49,2018-08-08 08:55:23,2018-08-08 13:50:00,2018-08-17 18:06:29,2018-09-04 00:00:00
3,949d5b44dbf5de918fe9c16f97b45f8a,f88197465ea7920adcdbec7375364d82,delivered,2017-11-18 19:28:06,2017-11-18 19:45:59,2017-11-22 13:39:59,2017-12-02 00:28:42,2017-12-15 00:00:00
4,ad21c59c0840e6cb83a9ceb5573f8159,8ab97904e6daea8866dbdbc4fb7aad2c,delivered,2018-02-13 21:18:39,2018-02-13 22:20:29,2018-02-14 19:46:34,2018-02-16 18:17:02,2018-02-26 00:00:00


In [11]:
#Struktura tabulky objednávek
orders.info()

<class 'pandas.DataFrame'>
RangeIndex: 99441 entries, 0 to 99440
Data columns (total 8 columns):
 #   Column                         Non-Null Count  Dtype
---  ------                         --------------  -----
 0   order_id                       99441 non-null  str  
 1   customer_id                    99441 non-null  str  
 2   order_status                   99441 non-null  str  
 3   order_purchase_timestamp       99441 non-null  str  
 4   order_approved_at              99281 non-null  str  
 5   order_delivered_carrier_date   97658 non-null  str  
 6   order_delivered_customer_date  96476 non-null  str  
 7   order_estimated_delivery_date  99441 non-null  str  
dtypes: str(8)
memory usage: 6.1 MB


In [12]:
#Chybějící hodnoty v tabulce objednávek
orders.isna().sum()

order_id                            0
customer_id                         0
order_status                        0
order_purchase_timestamp            0
order_approved_at                 160
order_delivered_carrier_date     1783
order_delivered_customer_date    2965
order_estimated_delivery_date       0
dtype: int64

In [13]:
#Výskyty každé hodnoty v sloupci order_status
orders["order_status"].value_counts()

order_status
delivered      96478
shipped         1107
canceled         625
unavailable      609
invoiced         314
processing       301
created            5
approved           2
Name: count, dtype: int64

In [14]:
customers = pd.read_csv(DATA / "C:\\Users\\svobo\\Projekty\\olist-analyza\\data\\raw\\olist_customers_dataset.csv")
customers.head()

,customer_id,customer_unique_id,customer_zip_code_prefix,customer_city,customer_state
0,06b8999e2fba1a1fbc88172c00ba8bc7,861eff4711a542e4b93843c6dd7febb0,14409,franca,SP
1,18955e83d337fd6b2def6b18a428ac77,290c77bc529b7ac935b93aa66c333dc3,9790,sao bernardo do campo,SP
2,4e7b3e00288586ebd08712fdd0374a03,060e732b5b29e8181a18229c7b0b2b5e,1151,sao paulo,SP
3,b2b6027bc5c5109e529d4dc6358b12c3,259dac757896d24d7702b9acbbff3f3c,8775,mogi das cruzes,SP
4,4f2d8ab171c80ec8364f7c12e35b23ad,345ecd01c38d18a9036ed96c73b8d066,13056,campinas,SP


In [15]:
customers.info()

<class 'pandas.DataFrame'>
RangeIndex: 99441 entries, 0 to 99440
Data columns (total 5 columns):
 #   Column                    Non-Null Count  Dtype
---  ------                    --------------  -----
 0   customer_id               99441 non-null  str  
 1   customer_unique_id        99441 non-null  str  
 2   customer_zip_code_prefix  99441 non-null  int64
 3   customer_city             99441 non-null  str  
 4   customer_state            99441 non-null  str  
dtypes: int64(1), str(4)
memory usage: 3.8 MB


In [16]:
customers.isna().sum()

customer_id                 0
customer_unique_id          0
customer_zip_code_prefix    0
customer_city               0
customer_state              0
dtype: int64

In [17]:
customers.value_counts()

customer_id                       customer_unique_id                customer_zip_code_prefix  customer_city          customer_state
06b8999e2fba1a1fbc88172c00ba8bc7  861eff4711a542e4b93843c6dd7febb0  14409                     franca                 SP                1
18955e83d337fd6b2def6b18a428ac77  290c77bc529b7ac935b93aa66c333dc3  9790                      sao bernardo do campo  SP                1
4e7b3e00288586ebd08712fdd0374a03  060e732b5b29e8181a18229c7b0b2b5e  1151                      sao paulo              SP                1
b2b6027bc5c5109e529d4dc6358b12c3  259dac757896d24d7702b9acbbff3f3c  8775                      mogi das cruzes        SP                1
4f2d8ab171c80ec8364f7c12e35b23ad  345ecd01c38d18a9036ed96c73b8d066  13056                     campinas               SP                1
                                                                                                                                      ..
17ddf5dd5d51696bb3d7c6291687be6f  1a29b476fee2

In [18]:
items = pd.read_csv(DATA / "C:\\Users\\svobo\\Projekty\\olist-analyza\\data\\raw\\olist_order_items_dataset.csv")
items.head()

,order_id,order_item_id,product_id,seller_id,shipping_limit_date,price,freight_value
0,00010242fe8c5a6d1ba2dd792cb16214,1,4244733e06e7ecb4970a6e2683c13e61,48436dade18ac8b2bce089ec2a041202,2017-09-19 09:45:35,58.90,13.29
1,00018f77f2f0320c557190d7a144bdd3,1,e5f2d52b802189ee658865ca93d83a8f,dd7ddc04e1b6c2c614352b383efe2d36,2017-05-03 11:05:13,239.90,19.93
2,000229ec398224ef6ca0657da4fc703e,1,c777355d18b72b67abbeef9df44fd0fd,5b51032eddd242adc84c38acab88f23d,2018-01-18 14:48:30,199.00,17.87
3,00024acbcdf0a6daa1e931b038114c75,1,7634da152a4610f1595efa32f14722fc,9d7a1d34a5052409006425275ba1c2b4,2018-08-15 10:10:18,12.99,12.79
4,00042b26cf59d7ce69dfabb4e55b4fd9,1,ac6c3623068f30de03045865e4e10089,df560393f3a51e74553ab94004ba5c87,2017-02-13 13:57:51,199.90,18.14


In [19]:
products = pd.read_csv(DATA / "C:\\Users\\svobo\\Projekty\\olist-analyza\\data\\raw\\olist_products_dataset.csv")
products.head()

,product_id,product_category_name,product_name_lenght,product_description_lenght,product_photos_qty,product_weight_g,product_length_cm,product_height_cm,product_width_cm
0,1e9e8ef04dbcff4541ed26657ea517e5,perfumaria,40.0,287.0,1.0,225.0,16.0,10.0,14.0
1,3aa071139cb16b67ca9e5dea641aaa2f,artes,44.0,276.0,1.0,1000.0,30.0,18.0,20.0
2,96bd76ec8810374ed1b65e291975717f,esporte_lazer,46.0,250.0,1.0,154.0,18.0,9.0,15.0
3,cef67bcfe19066a932b7673e239eb23d,bebes,27.0,261.0,1.0,371.0,26.0,4.0,26.0
4,9dc1a7de274444849c219cff195d0b71,utilidades_domesticas,37.0,402.0,4.0,625.0,20.0,17.0,13.0


In [21]:
pd.crosstab(
    orders["order_status"],
    orders["order_delivered_customer_date"].isna(),
    margins=True
)

order_delivered_customer_date,False,True,All
order_status,,,
approved,0,2,2
canceled,6,619,625
created,0,5,5
delivered,96470,8,96478
invoiced,0,314,314
processing,0,301,301
shipped,0,1107,1107
unavailable,0,609,609
All,96476,2965,99441


In [22]:
pd.crosstab(
    orders["order_status"],
    orders["order_approved_at"].isna(),
    margins=True
)

order_approved_at,False,True,All
order_status,,,
approved,2,0,2
canceled,484,141,625
created,0,5,5
delivered,96464,14,96478
invoiced,314,0,314
processing,301,0,301
shipped,1107,0,1107
unavailable,609,0,609
All,99281,160,99441


In [23]:
pd.crosstab(
    orders["order_status"],
    orders["order_delivered_carrier_date"].isna(),
    margins=True
)

order_delivered_carrier_date,False,True,All
order_status,,,
approved,0,2,2
canceled,75,550,625
created,0,5,5
delivered,96476,2,96478
invoiced,0,314,314
processing,0,301,301
shipped,1107,0,1107
unavailable,0,609,609
All,97658,1783,99441


Dosavadní zjištění a úvahy:
A - Nejvíce řádků obsahuje tabulka geolokace: olist_geolocation_dataset.csv: 1000163 rows, 5 columns, obsahující směrovací čísla a zeměpisnou délku a šířku místa.
B - Soubor Orders:Počty řádků s chybějícími daty:
order_approved_at                 160
order_delivered_carrier_date     1783
order_delivered_customer_date    2965
    approved_at: Hypotéza k ověření: proč nebyla objednávka schválena? 
        Metodou Crosstab se ukazuje, že datum schválení chybí u 141 "canceled" objednávek, které mohly být zrušeny krátce po vytvoření, 5 objednávek "created", které na schválení čekají a kupodivu 14 "delivered" orders, což by mohla být chyba v datech, 141+5+14=160
    delivered_carrier_date: Hypotéza k ověření: Jedná se "pouze" o kombinaci stavu před předáním přepravci nebo jsou zde i nějaké chyby v datech? 
        Crosstab: Approved 2 + Canceled 550 + Created 5 + Delivered 2 + Invoiced 314 + Processing 301 + Unuvailable 609 = 1783, tj. teoreticky chybou by mohly být 2 čádky stavu "delivered" - zde by mohla být lidská chyba zadání dat
    delivered_customer_date: Hypotéza k ověření: Jedná se opět pouze o kombinaci stavů nebo jsou zde i chyby v datech? Proč nebyly zásilky dodány?
        Crosstab: Approved 2 + Canceled 619 + Created 5 + Delivered 8 + Invoiced 314 + Processing 301 + Shipped 1107 + Unavailable 609, tj. diskutabilní jsou chybějící data u stavu Delivered 


